stacks-signer/src/config.rs

```diff
@@ -338,19 +338,20 @@ pub fn build_signer_config_tomls(
     timeout: Option<Duration>,
     network: &Network,
     password: &str,
+    mut port_start: usize,
 ) -> Vec<String> {
     let mut signer_config_tomls = vec![];
 
-    let mut port = 30000;
-    let run_stamp = rand::random::<u16>();
-    let db_dir = format!(
-        "/tmp/stacks-node-tests/integrations-signers/{:#X}",
-        run_stamp,
-    );
-    fs::create_dir_all(&db_dir).unwrap();
-    for (ix, stacks_private_key) in stacks_private_keys.iter().enumerate() {
-        let endpoint = format!("localhost:{}", port);
-        port += 1;
+    for stacks_private_key in stacks_private_keys {
+        let endpoint = format!("localhost:{}", port_start);
+        port_start += 1;
+
+        let stacks_public_key = StacksPublicKey::from_private(stacks_private_key).to_hex();
+        let db_dir =
+            format!("/tmp/stacks-node-tests/integrations-signers/signer_{stacks_public_key}");
+        let db_path = format!("{db_dir}/signerdb.sqlite");
+        fs::create_dir_all(&db_dir).unwrap();
```

Comment:
You should remove the directory before creating it, in case the same private keys get reused across invocations.  Otherwise you'd have some surprising test behavior on the second and subsequent runs.

Reply:
Right -- this is why I suggested using a run stamp. The way that this is setup now, test runs will step on each other and use the same db_path. The run stamp prevents that from happening, and also lets you access the prior run's state for debugging.

For the concern that you actually need this to be deterministic, I'm not sure that I agree -- if you just generate the configs at the start of the test, when you restart a signer, you can just pass it the same config it started with and it will have the same db.

Reply:
Sure, allowing private keys to get reused across invocations was my intention here. But I understand that other integration tests might hard-code private keys or use seeded random numbers so I will re-introduce the run stamp. This will require us to store the signer config in the `SignerTest` - but that shouldn't be any problem.

Reply:
Done :heavy_check_mark: